spring-kafka/src/main/java/org/springframework/kafka/listener/KafkaMessageListenerContainer.java

```diff
@@ -1449,34 +1425,10 @@ public String toString() {
 
 		private final class ConsumerBatchAcknowledgment implements Acknowledgment {
 
-			private final List<ConsumerRecord<K, V>> records;
-
-			ConsumerBatchAcknowledgment(List<ConsumerRecord<K, V>> records) {
-				// make a copy in case the listener alters the list
-				this.records = new LinkedList<ConsumerRecord<K, V>>(records);
-			}
-
-			@Override
-			public void acknowledge() {
-				Assert.state(ListenerConsumer.this.isAnyManualAck,
-						"A manual ackmode is required for an acknowledging listener");
-				for (ConsumerRecord<K, V> record : getHighestOffsetRecords(this.records)) {
-					processAck(record);
-				}
-			}
-
-			@Override
-			public String toString() {
-				return "Acknowledgment for " + this.records;
-			}
-
-		}
-
-		private final class ConsumerRecordsAcknowledgment implements Acknowledgment {
-
 			private final ConsumerRecords<K, V> records;
 
-			ConsumerRecordsAcknowledgment(ConsumerRecords<K, V> records) {
+			ConsumerBatchAcknowledgment(ConsumerRecords<K, V> records) {
+				// make a copy in case the listener alters the list
```

Comment:
??? But you don't make copy here.
I guess this is a leftover from the previous version.

Anyway let's think what really makes sense to to do here!